web_ui/packages/ui/src/virtualize-list-layout/virtualize-list-layout.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2022-2025 Intel Corporation
+// LIMITED EDGE SOFTWARE DISTRIBUTION LICENSE
+
+import { ReactNode, useRef } from 'react';
+
+import { View } from '@adobe/react-spectrum';
+import { useLoadMore } from '@react-aria/utils';
+import {
+    ListBox as AriaComponentsListBox,
+    ListBoxItem,
+    ListLayout,
+    ListLayoutOptions,
+    Selection,
+    Virtualizer,
+} from 'react-aria-components';
+
+import { LoadingIndicator } from '../loading/loading-indicator.component';
+
+import classes from './virtualize-list-layout.module.scss';
+
+interface VirtualizedListLayoutProps<T> {
+    items: T[];
+    selected: Selection;
+    isLoading: boolean;
+    ariaLabel?: string;
+    layoutOptions: ListLayoutOptions;
+    renderLoading?: () => ReactNode;
+    renderItem: (item: T) => ReactNode;
+    onLoadMore: () => void;
+}
+
+export const VirtualizedListLayout = <T extends { id: string; name: string }>({
+    items,
+    isLoading,
+    selected,
+    ariaLabel,
+    layoutOptions,
+    renderLoading = () => <LoadingIndicator size={'M'} />,
```

Comment:
I'm wondering if it would not be better to do `renderLoading = <LoadingIndicator />`, same for `renderItem` thsi could be a `Item` that is passed to the virtualized list layout.

One primary downside of doing `item={() => <SomeItem />` is that it redefines `item` every time the parent component gets rerendered - which then results in remounting all item components.

Reply:
Discussed offline. We’ll keep the prop-based rendering since the virtualized list already applies optimizations to the rendered items.

Reply:
Discussed offline: it doesn't look like it impacts the component's state (i.e. an update from say the status or jobs endpoint won't trigger a rerender that then remounts the component and then resets the state) so let's keep it as is.